src/parse/mod.rs

```diff
@@ -415,9 +423,10 @@ mod tests {
         })
     }
     #[inline]
-    pub(crate) fn parser(input: &str) -> Parser<Lexer> {
-        let lex = Lexer::new("<test suite>".to_string(), input.chars(), false);
-        Parser::new(lex, false).unwrap()
+    pub(crate) fn parser(input: &str) -> Parser<impl Iterator<Item = Lexeme>> {
+        let tokens =
+            Lexer::new("<test suite>".to_string(), input.chars(), false).collect::<Vec<_>>();
+        Parser::new(tokens.into_iter(), false).unwrap()
```

Comment:
Why did you make this change? Now that they're using different error handlers there shouldn't be a problem with lifetimes.

Reply:
This is changed to collect the tokens into a Vec to match the way it happens in the main compile function. Should I change it back to using the lexer directly?

Reply:
Yes, I don't see a reason to change it.

Reply:
I changed this because of how the main compile function collects into a `Vec` then passes the `Vec` to parser. This no longer has the lifetime issues that I originally made the change for, but the reasoning still stands. Should I change it back?

Reply:
Ok, now can you change all the others back? e.g. https://github.com/jyn514/rcc/pull/164/files#diff-b4aea3e418ccdb71239b96952d9cddb6R69